src/cmakeTaskProvider.ts

```diff
@@ -39,23 +46,30 @@ export class CMakeTaskProvider implements vscode.TaskProvider {
         }
     }
 
-    public updateDefaultTargets(defaultTargets: string[] | undefined) {
+    public updateDefaultTargets(defaultTargets?: string[]) {
         this.defaultTargets = (defaultTargets && defaultTargets.length > 0) ? defaultTargets :
             this.cmakeDriver ? [this.cmakeDriver.allTargetName] : [CMakeTaskProvider.allTargetName];
     }
 
     public async provideTasks(): Promise<CMakeTask[]> {
         // Create a CMake build task
         const result: CMakeTask[] = [];
-        const taskName: string = "CMake: build";
+        this.updateDefaultTargets();
+        // Provide build task.
+        const taskName: string = CommandType.build;
         const definition: CMakeTaskDefinition = {
             type: CMakeTaskProvider.CMakeScriptType,
-            label: taskName,
-            command: "build"
+            label: CMakeTaskProvider.CMakeSourceStr + ": " + taskName,
```

Comment:
I don't think this will have the necessary localization.  It looks like you only provide the build task now, but what about the others?  If so, we might want to use the same strings from package.nls.json.  I don't think the `:` separator is needed if we reuse those.

Reply:
(and by "reuse", I mean copy the strings in here. We can't use those localizations directly)

Reply:
the reason I used the "Cmake:" at the beginning of the name is to be consistent with what other extensions are doing so that the user will know which extension is providing this task.
![image](https://user-images.githubusercontent.com/16143955/161296950-caa372a3-4345-47ea-b1b6-e405d31fe326.png)
For example this is how the user will see the task in the lists of the provided tasks.

Reply:
I will fix the localization though.